### PR TITLE
Issue #11307- Deprecate and remove processorhelper.NewUnhealthyProcessorCreateSettings

### PR DESCRIPTION
**Pull request by olamideTiana**
Deprecate and remove NewUnhealthyProcessorCreateSettings

Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/11307

### CODE REVIEW

**Comment by olamideTiana** [comment]
fixings made

**Review by olamideTiana — COMMENTED** [review]
Done

File: processor/processortest/unhealthy_processor.go [hunk 1]
```diff
@@ -14,7 +14,8 @@ import (
 	"go.opentelemetry.io/collector/processor"
 )
 
-// NewUnhealthyProcessorCreateSettings returns a new nop settings for Create* functions.
+// Deprecated: NewUnhealthyProcessorCreateSettings returns a new nop settings for Create* functions. To be removed in future.
```
Comment (bogdandrutu) [inline processor/processortest/unhealthy_processor.go:17]: Please follow the convention here: https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#breaking-changes by adding the version.

Please include a changelog entry

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,11 +14,13 @@ If you are looking for developer-facing changes, check out [CHANGELOG-API.md](./
 - `service/telemetry`: Change default metrics address to "localhost:8888" instead of ":8888" (#11251)
   This behavior can be disabled by disabling the feature gate 'telemetry.UseLocalHostAsDefaultMetricsAddress'.
 - `loggingexporter`: Removed the deprecated logging exporter.  Use the debug exporter instead. (#11037)
+- `processortest` : Removed and deprecated NewUnhealthyProcessorCreateSettings. Use NewNopSettings instead. (#11307)
```
Comment (bogdandrutu) [inline CHANGELOG.md:17]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,11 +14,13 @@ If you are looking for developer-facing changes, check out [CHANGELOG-API.md](./
...
 
 ### 🚩 Deprecations 🚩
 
 - `service/telemetry`: Deprecate service::telemetry::metrics::address in favor of service::telemetry::metrics::readers. (#11205)
 - `processorhelper`: Deprecate BuildProcessorMetricName as it's no longer needed since introduction of mdatagen (#11302)
+- `processortest` : Deprecate NewUnhealthyProcessorCreateSettings as it's not used and equivalent to NewNopSettings (#11307)
```
Comment (bogdandrutu) [inline CHANGELOG.md:23]:
```suggestion
```

**Review by bogdandrutu — APPROVED** [review]

**Comment by codeboten** [comment]
_update the description to automatically close the related issue_

File: processor/processortest/unhealthy_processor.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 	"go.opentelemetry.io/collector/processor"
 )
 
-// NewUnhealthyProcessorCreateSettings returns a new nop settings for Create* functions.
+// Deprecated: [v.0.111.0] not used. To be removed. Use NewNopSettings instead.
```
Comment (codeboten) [inline processor/processortest/unhealthy_processor.go:17]:
```suggestion
// Deprecated: [v0.112.0] not used. To be removed. Use NewNopSettings instead.
```

File: .chloggen/processortest-deprecated-removed-func.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (codeboten) [inline .chloggen/processortest-deprecated-removed-func.yaml:4]:
```suggestion
change_type: deprecation
```

File: .chloggen/processortest-deprecated-removed-func.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: processortest
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecated and removed 'NewUnhealthyProcessorCreateSettings'. Use NewNopSettings instead.
```
Comment (codeboten) [inline .chloggen/processortest-deprecated-removed-func.yaml:10]:
```suggestion
note: Deprecated 'NewUnhealthyProcessorCreateSettings'. Use NewNopSettings instead.
```

**Review by codeboten — COMMENTED** [review]
the comment and changelog needs updating

**Review by songy23 — APPROVED** [review]

**Pull request merged**